google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/BigtableCloudMonitoringExporter.java

```diff
@@ -208,4 +233,62 @@ public CompletableResultCode shutdown() {
   public AggregationTemporality getAggregationTemporality(InstrumentType instrumentType) {
     return AggregationTemporality.CUMULATIVE;
   }
+
+  private CompletableResultCode exportGceOrGkeMetrics(Collection<MetricData> collection) {
+    CompletableResultCode gceOrGkeExportCode = new CompletableResultCode();
+    if (gceOrGkeResource == null) {
+      return gceOrGkeExportCode.succeed();
+    }
+    List<TimeSeries> gceOrGceTimeSeries;
+    try {
+      gceOrGceTimeSeries =
+          BigtableExporterUtils.convertToGceOrGkeTimeSeries(
+              collection.stream()
+                  .filter(
+                      md ->
+                          md.getName()
+                              .contains(BuiltinMetricsConstants.PER_CONNECTION_ERROR_COUNT_NAME))
+                  .collect(Collectors.toList()),
+              taskId,
+              gceOrGkeResource);
+    } catch (Throwable e) {
+      logger.log(
+          Level.WARNING,
+          "Failed to convert per connection error count data to cloud monitoring timeseries.",
+          e);
+      return CompletableResultCode.ofFailure();
+    }
+
+    ProjectName gceOrGkeProjectName =
+        ProjectName.of(gceOrGkeResource.getLabelsOrThrow(GCE_OR_GKE_PROJECT_ID_LABEL));
+    CreateTimeSeriesRequest gceOrGkeRequest =
+        CreateTimeSeriesRequest.newBuilder()
+            .setName(gceOrGkeProjectName.toString())
+            .addAllTimeSeries(gceOrGceTimeSeries)
+            .build();
+
+    ApiFuture<Empty> gceOrGkeFuture =
+        this.client.createServiceTimeSeriesCallable().futureCall(gceOrGkeRequest);
+
+    ApiFutures.addCallback(
```

Comment:
I think it would cleaner to return the ApiFuture and then have helper to compose multiple futures into a single code